scrapers_next/nd/people.py

```diff
@@ -1,118 +1,160 @@
-from spatula import URL, CSS, HtmlListPage, HtmlPage, XPath, SelectorError
-from openstates.models import ScrapePerson
 import re
+import attr
+from spatula import HtmlPage, HtmlListPage, XPath, CSS, SelectorError
+from openstates.models import ScrapePerson
+
+
+class PaginationError(Exception):
+    pass
+
+
+class PartyError(Exception):
+    pass
+
+
+@attr.s(auto_attribs=True)
+class PartialMember:
+    image: str
+    name: str
+    chamber: str
+    district: str
+    party: str
+    url: str
+    start_date: str = ""
 
 
 class LegDetail(HtmlPage):
+    example_source = "https://www.ndlegis.gov/biography/mary-adams"
+
     def process_page(self):
-        p = self.input
+        name_list = self.input.name.split()
+        given_name = name_list.pop(0)
```

Comment:
Worth documenting why we're popping a random magic number.

Reply:
Thank you for pointing this out, because upon review I realized I never needed to pop() at all. I went ahead and [changed to an indexing approach](https://github.com/openstates/openstates-scrapers/pull/4162/commits/ed81cdcef99f666fd2d8464c036fe5263b687dee#diff-b4eb1b6b6189a33b171b390670cb2f7c25db0b25acd7e596c8c5afb685d778a6R35-R37) for accessing  a member's `given_name` and `family_name`.

But after running this I also realized my approach for isolating `family_name` wasn't sufficient. While it excluded middle initials like "B." and suffixes like "Jr.", in the cases where someone had a name with a suffix (i.e. "Howard C. Anderson, Jr." it accidentally captured the comma as part of the family_name so it was getting stored as "Anderson,"
- the newly committed line 37 eliminates any such commas.